src/ShaderSwapper.Core/ShaderSwapper.cs

```diff
@@ -30,62 +30,190 @@ public class ShaderSwapper : BaseUnityPlugin
+
+                    public List<string> Entries { get; set; } = new List<string>();
+
+                    public EntryList() { }
+
+                    public EntryList(IEnumerable<string> entries)
+                    {
+                        Entries = new List<string>(entries);
+                    }
+
+                    public EntryList(IEnumerable<XElement> collection)
+                    {
+                        Entries = new List<string>(collection.Select(e => e.Value));
+                    }
+
+                    public IEnumerable<XElement> ToElements()
+                    {
+                        return Entries.Select(e => new XElement(EntryElementName, e));
+                    }
+                }
+
+                public static string ElementName { get; } = "Rule";
+                public static string ShaderAttributeName { get; } = "Name";
+                public static string IncludeElementName { get; } = "Include";
+                public static string ExcludeElementName { get; } = "Exclude";
+
+                public string Shader { get; set; } = "MISSING SHADER NAME";
+                public EntryList Include { get; set; } = new EntryList();
+                public EntryList Exclude { get; set; } = new EntryList();
+
+                public Rule(string shader)
+                {
+                    Shader = shader;
+                }
+
+                public Rule(string shader, IEnumerable<string> include, IEnumerable<string> exclude)
+                {
+                    Shader = shader;
+                    Include = new EntryList(include);
+                    Exclude = new EntryList(exclude);
+                }
+
+                public Rule(XElement element)
+                {
+                    if (element == null || element.Name != ElementName)
+                        return;
+
+                    if (element.Attribute(ShaderAttributeName) != null)
+                        Shader = element.Attribute(ShaderAttributeName).Value;
+                    if (element.Element(IncludeElementName) != null)
+                        Include = new EntryList(element.Element(IncludeElementName).Elements());
+                    if (element.Element(ExcludeElementName) != null)
+                        Exclude = new EntryList(element.Element(ExcludeElementName).Elements());
+                }
+
+                public XElement ToElement()
+                {
+                    var element = new XElement(ElementName, new XAttribute(ShaderAttributeName, Shader));
+                    if (Include.Entries.Count > 0)
+                        element.Add(new XElement(IncludeElementName, Include.ToElements()));
+                    if (Exclude.Entries.Count > 0)
+                        element.Add(new XElement(ExcludeElementName, Exclude.ToElements()));
+                    return element;
+                }
+
+                public bool TryGetToShader(Material material, out string toShader)
```

Comment:
How about naming it "TryGetReplacementShader"? Or just add a summary to the methods to explain what they do.